src/app/reporting/Engine.cpp

```diff
@@ -1004,7 +1005,21 @@ void Engine::ScheduleUrgentEventDeliverySync(Optional<FabricIndex> fabricIndex)
     Run();
 }
 
-}; // namespace reporting
+void Engine::MarkDirty(const ConcreteAttributePath & path)
+{
+    // NOTE: original MatterReportingAttributeChangeCallback ALSO does a
+    //       IncreaseClusterDataVersion. However that ties directly to ember so it is not
```

Comment:
So what increases the data versions, then?  Or is that handled by the data model provider directly?

Reply:
It was supposed to be a TODO on my side - this is why some integration tests fail I believe.

I will add this increase inside other places (generally ember is supposed to do this internally: when an attribute is changed, it should increase the version it holds and not rely externally on the version management).

Reply:
I added a change to propagate the change status from within ember, so that the codegen data model can actually match the dirty marking from ember.